pkg/apis/compliance/v1alpha1/customrule_types.go

```diff
@@ -0,0 +1,114 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ScannerType string
+
+const (
+	ScannerTypeCEL      ScannerType = "CEL"
+	ScannerTypeOpenSCAP ScannerType = "OpenSCAP"
+	ScannerTypeUnknown  ScannerType = "Unknown"
+)
+
+type InputPayload struct {
+	// The kubernetes resource that will be used as input
+	// +nullable
+	// +optional
+	KubeResource `json:",inline,omitempty"`
+}
+
+type InputResourceType string
+
+const (
+	InputResourceTypeKubeResource InputResourceType = "KubeGroupVersionResource"
+	InputResourceTypeUnknown      InputResourceType = "Unknown"
+)
+
+// ResourceInput defines a Kubernetes resource that will be fetched for CEL evaluation
+type KubeResource struct {
+	// Name is the variable name used to reference this resource in the CEL expression
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	Name string `json:"name"`
+	// Type is the type of the resource
+
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=KubeGroupVersionResource
+	Type InputResourceType `json:"type"`
+
+	// APIGroup is the Kubernetes API group of the resource
+	// +kubebuilder:validation:Required
+	APIGroup string `json:"apiGroup"`
+
+	// Version is the Kubernetes API version of the resource
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	Version string `json:"version"`
+
+	// Resource is the Kubernetes resource type
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	Resource string `json:"resource"`
+
+	// Namespace is the Kubernetes namespace of the resource
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
+}
+
+type CELPayload struct {
+
+	// ScannerType specifies what type of check this rule performs
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=CEL
+	ScannerType ScannerType `json:"scannerType"`
```

Comment:
I wonder if `ScannerType` should be defined outside the `CELPayload`. What happens if/when we add another implementation along side CEL? Will we need to define `ScannerType` in its payload, too?

Reply:
that's a really good point!